### PR TITLE
*: use stdlib context instead of golang.org/x/net/context

### PR DESCRIPTION
**Pull request by simonpasquier**
Attempt to close https://github.com/grpc-ecosystem/go-grpc-prometheus/issues/69. This change also introduces support for Go modules.

cc @brancz @bwplotka

### CODE REVIEW

**Comment by brancz** [comment]
lgtm :+1: giving @bwplotka the last call

**Comment by simonpasquier** [comment]
No hurry, I'll ping you again when Travis is happy...

File: .travis.yml [hunk 1]
```diff
@@ -1,25 +1,21 @@
...
 
+env:
+- GO111MODULE=on
+
 install:
-  - go get github.com/prometheus/client_golang/prometheus
-  - go get google.golang.org/grpc
-  - go get golang.org/x/net/context
-  - go get github.com/stretchr/testify
+- if [[ $TRAVIS_GO_VERSION =~ ^1\.(7|8|9|10)\. ]]; then go get github.com/prometheus/client_golang/prometheus google.golang.org/grpc golang.org/x/net/context github.com/stretchr/testify; fi
```
Comment (bwplotka) [inline .travis.yml]: why this if? Worth a comment? (:

File: server_test.go [hunk 2]
```diff
@@ -52,6 +52,7 @@ type ServerInterceptorTestSuite struct {
 	clientConn     *grpc.ClientConn
 	testClient     pb_testproto.TestServiceClient
 	ctx            context.Context
+	cancel         context.CancelFunc
```
Comment (bwplotka) [inline server_test.go]: is this used anywhere? It should be atleased emptied (make sure to do `cancel()` at the tear down)

**Comment by bwplotka** [comment]
Travis is not happy though:
```
22.17s$ if [[ $TRAVIS_GO_VERSION =~ ^1\.(7|8|9|10)\. ]]; then go get github.com/prometheus/client_golang/prometheus google.golang.org/grpc golang.org/x/net/context github.com/stretchr/testify; fi
# google.golang.org/grpc/credentials/internal
../../../google.golang.org/grpc/credentials/internal/syscallconn.go:29: syntax error: unexpected = in type declaration
# google.golang.org/grpc/internal/syscall
../../../google.golang.org/grpc/internal/syscall/syscall_linux.go:79: tcpconn.SyscallConn undefined (type *net.TCPConn has no field or method SyscallConn)
../../../google.golang.org/grpc/internal/syscall/syscall_linux.go:100: tcpconn.SyscallConn undefined (type *net.TCPConn has no field or method SyscallConn)
# google.golang.org/grpc/resolver/dns
../../../google.golang.org/grpc/resolver/dns/dns_resolver.go:86: unknown field 'Dial' in struct literal of type net.Resolver
# golang.org/x/net/http2
../../../golang.org/x/net/http2/server.go:220: s.RegisterOnShutdown undefined (type *http.Server has no field or method RegisterOnShutdown)
The command "if [[ $TRAVIS_GO_VERSION =~ ^1\.(7|8|9|10)\. ]]; then go get github.com/prometheus/client_golang/prometheus google.golang.org/grpc golang.org/x/net/context github.com/stretchr/testify; fi" failed and exited with 2 during .
```

**Comment by simonpasquier** [comment]
Tests fail for Go 1.7 and 1.8 because 4 months ago, `google.golang.org/grpc` removed support for these versions (see https://github.com/grpc/grpc-go/pull/2428). IIUC `go-grpc-prometheus` would need to align of this?

**Pull request merged**